Home-task-5/p2.js

```diff
@@ -0,0 +1,12 @@
+//A Chain adding function
+// http://www.codewars.com/kata/a-chain-adding-function
+
+function add(num1) {
+  var fn = function (num2) {
+    return add(num1 + num2);      //recursive call
+  }
+  fn.valueOf = function () {  //obj to primitive
+    return num1;
+  }
+  return fn;
+}
```

Comment:
As specified in problem, result of add(1)(2)(3); // 6 
but it will always return a function until we use valueOf but I don't think that's expected.

Reply:
Everytime at last, valueOf would be invoked. So it would be satisying all the test cases of given problem 

> As specified in problem, result of add(1)(2)(3); // 6 but it will always return a function until we use valueOf but I don't think that's expected.